python/ray/_private/runtime_env/conda.py

```diff
@@ -320,32 +319,25 @@ async def create(
         context: RuntimeEnvContext,
         logger: logging.Logger = default_logger,
     ) -> int:
+        if not runtime_env.has_conda():
+            return 0
+
         def _create():
-            if uri is None:
-                # The "conda" field is the name of an existing conda env, so no
-                # need to create one.
-                # TODO(architkulkarni): Try "conda activate" here to see if the
-                # env exists, and raise an exception if it doesn't.
-                conda_env_name = runtime_env.get("conda")
-                if conda_env_name is None or not isinstance(conda_env_name, str):
-                    raise ValueError(
-                        "Unexpected conda environment was specified "
-                        f"from a runtime env. {runtime_env}. It must "
-                        "be an existing env, a yaml file, or a "
-                        "corresponding dict."
-                    )
+            result = parse_and_validate_conda(runtime_env.get("conda"))
```

Comment:
Is it necessary to parse and validate again in the agent? I don't think we do it for other the other runtime env fields, it might cause some issues.

I think we just need to parse the URI here. We can do it completely analogous to `pip.py` if I understand correctly.  The one potentially tricky thing is deletion:  For `conda`, in the case of an *existing env* (the case considered by this PR), we never want to delete the env for garbage collection. In that sense, we would somehow need to track the "no delete" property in the URI itself, or maybe just revert to the previous approach (sorry for not realizing this earlier...) Any thoughts about this?

For testing, you can glance over the tests in `test_runtime_env_conda_and_pip_*`.

Reply:
there's no deletion happening right? The URI folder will just be empty.

Reply:
are you saying it will raise an exception because the prefix doesn't exist?

Reply:
> Is it necessary to parse and validate again in the agent? I don't think we do it for other the other runtime env fields, it might cause some issues.

Are you suggesting to just use runtime_env.get("conda") to check if the value is string instead?

Reply:
For testing, there are already tests for GC behavior in `test_runtime_env_conda_and_pip*.py` and there are basic tests for the existing conda envs in `test_runtime_env_complicated`. There is a fixture here called `conda_envs` which sets up existing conda envs.  

We don't have an e2e test for the unintended GC scenario mentioned in my comment above, it might make sense to add one (to verify an existing env isn't deleted).  Up to you though

Reply:
> > Is it necessary to parse and validate again in the agent? I don't think we do it for other the other runtime env fields, it might cause some issues.
> > 
> Are you suggesting to just use runtime_env.get("conda") to check if the value is string instead?
Yeah, I think that would be slightly cleaner.  Because `runtime_env.get("conda") ` should already contain the output of `parse_and_validate` which was called on the client side. It's a nit though.